### ISSUE TITLE
Collision Inconsisency on Sides of Screen

### ISSUE THREAD

**Issue by SetsuneW** [chrismaltby/gb-studio#315]
**Describe the bug**
The Collision Masking does not work consistently between the sides of the screen. In some cases it is desirable to put the masking on only one or two extreme edges of the map in order to avoid wasting usable player space. This can be accomplished by masking just one side of the map where it would normally wrap.

Unfortunately, the engine handles this differently. For example, if a single vertical strip is placed on the extreme left, the camera and player will not wrap over when pressing against the right side; working as expected. The same goes for a strip across the bottom, pressing against the top will not wrap the camera or player. But when placed along the top, the camera and apparently player will wrap around when pressing against the bottom, although they will be obstructed unless they walk around it. When placed at the bottom and walking up through the top, the camera will move down and the player will seemingly warp past the collision.

So depending on what side of the screen you place them on, Collision Masks act 3 different ways.

**To Reproduce**
1. Create a map of presumably any size, probably bigger than 1 screen in width and/or height.
2. Place a single strip of Collision masking along the side you want to test.
3. Walk to the opposite side of the map and attempt to keep walking off-screen.

**Expected behavior**
A single consistent behavior no matter where the collision is located. Preferably the Left/Bottom behavior, where the game will check for collision on the opposite end before wrapping the player and camera.

Alternatively, it would be a nice feature to automatically set a barrier on the outer perimeter of maps, so that no tile space has to be wasted.

**Platform (please complete the following information):**
 - OS: Windows 10 Version 1903 (Build 18362,418)
 - App version: 1.2.0-alpha.1 (commit 3ef5be3 of the Develop branch)

**Comment by chrismaltby** [comment]
Hi @SetsuneW I've actually gone with your alternative option, making it so actors can't walk outside the scene edges so you don't need to add collisions around the scene perimeter at all and can use the full scene area for walking.

Need to do some more testing before I can merge but my progress so far is in PR #322

**Comment by SetsuneW** [comment]
Sounds good! Will this prevent actors from being forced to walk "off-screen" via script? (To make an NPC leave a scene naturally while on a map edge.)

**Comment by chrismaltby** [comment]
I think walking off-screen is something that was quite inconsistent in a similar way, on a 256px map it wouldn't have been possible and on a smaller maps it would have only really worked on the right and bottom edges. One work around though would be to walk the NPC right up to the edge of scene and then immediately use an "Actor: Hide" to remove them.

**Comment by SetsuneW** [comment]
That is true, although setting a relative offset seemed to work in my experimentation, but had a few quirks. One other potential, just to float: Setting a default offset for smaller than max maps, so they don't start at 0,0? But I suppose you could just pack in extra border tiles to the map image for the same effect.
